pkg/controller/installation/core_controller.go

```diff
@@ -767,8 +763,8 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		return reconcile.Result{}, err
 	}
 
-	terminating := (instance.DeletionTimestamp != nil)
-	if terminating {
+	installationDeleted := (instance.DeletionTimestamp != nil)
```

Comment:
I don't like this name change, if the installation was deleted then this assignment would be accessing invalid memory and the assignment value could be changed to `instance == nil`.

Reply:
I also don't think it is accurate, as it isn't deleted yet. I think `terminating` is terse but descriptive enough but if you don't think so then I'd suggest something like `installationTerminating` or `installationCleanup`.

Reply:
@tmjd strictly speaking, I think this is actually accurate - the Installation has had a "delete" API action called against it and it succeeded. The presence of a DeletionTimestamp implies that the object has been deleted.

That said, I can see why this is a confusing name. I think "terminating" is worse - it has always caused me to misinterpret this code because "terminating" to me implies the Pod API state in Kubernetes, so I always read this as "The operator pod is terminating", which is wrong.

I think the correct name for this is probably something like "installationFinalizing" or "finalizeInstallation" as that is what the code is doing when this field is "true". 

What do you think?

Reply:
I disagree that it is accurate. If the resource was deleted then it would be gone, it has a deletetion timestamp exactly because it hasn't been able to be deleted yet but there is an intention for it to be deleted.

I can see the confusion with the term "terminating".

I understand why you've included `finalize` in your name suggestion but I think that might be too generic, that could also mean we are setting the finalizers on the Installation. Also I don't know that in general, people will understand what "finalizing" would mean in a variable name because I don't know that finalizers are commonly understood.

How about `installationDeleting`? I think `installationCleanup` would be a viable option too.

Reply:
> people will understand what "finalizing" would mean in a variable name

I don't think it's necessarily a good idea to prioritize making this variable more understandable to people who don't understand what finalizers are. This code is fundamentally related to the process of finalizing the resource and we use finalizes extensively as part of this logic, so anyone trying to understand this logic better understand what finalizers are. Putting even a tiny bit of distance between the variable name and the mechanism it represents just makes the logic harder to understand IMO, as evidenced by the "terminating" confusion I've always encountered.

Reply:
@tmjd I went with `installationMarkedForDeletion` as a compromise.